src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -882,17 +882,21 @@ private bool ParallelProcessBlock(ProcessBlocksInfo round, Block block, ChainedH
 
                                 conn.Rollback();
 
+                                throw;
+                            }
+                            finally
+                            {
                                 // Ensure locks are released.
                                 this.EndBatch(round);
 
-                                throw;
+                                this.Metrics.ProcessTime += (DateTime.Now.Ticks - flagFall);
+                                this.Metrics.LogMetrics(this, conn, chainedHeader, wallet);
                             }
-
-                            this.Metrics.ProcessTime += (DateTime.Now.Ticks - flagFall);
-                            this.Metrics.LogMetrics(this, conn, chainedHeader, wallet);
                         }
-
-                        this.EndBatch(round);
+                        else
```

Comment:
What was the purpose of adding an `else` here?

Reply:
To avoid `EndBatch` being called twice.